comment/mixins.py

```diff
@@ -1,162 +1,143 @@
-from django.http import HttpResponseBadRequest, HttpResponseForbidden
-from django.views.generic import FormView
 
 from comment.conf import settings
-from comment.utils import is_comment_admin, is_comment_moderator
+from comment.utils import is_comment_admin, is_comment_moderator, can_block_user, can_moderate_flagging
 from comment.validators import ValidatorMixin
-from comment.messages import ErrorMessage, FlagError, FollowError
-from comment.service.email import DABEmailService
-from comment.messages import EmailInfo
-from comment.forms import CommentForm
-from comment.context import DABContext
-from comment.responses import DABResponseData
+from comment.messages import ErrorMessage, FlagError, FollowError, BlockedUserMSG
+from comment.responses import UTF8JsonResponse
+from comment.models import BlockedUser
 
 
 class AJAXRequiredMixin:
+    status = 403
+    reason = ErrorMessage.NON_AJAX_REQUEST
+
     def dispatch(self, request, *args, **kwargs):
         if not request.META.get('HTTP_X_REQUESTED_WITH', None) == 'XMLHttpRequest':
-            return HttpResponseBadRequest(ErrorMessage.NON_AJAX_REQUEST)
+            data = {'status': self.status, 'reason': self.reason}
+            return UTF8JsonResponse(status=self.status, data=data)
         return super().dispatch(request, *args, **kwargs)
 
 
 class BasePermission(AJAXRequiredMixin):
-    def has_permission(self, request):
-        return True
+    reason = ErrorMessage.NOT_AUTHORIZED
 
-    def has_object_permission(self, request, obj):
+    def has_permission(self, request):
         return True
 
-
-class BaseCommentMixin(LoginRequiredMixin, BasePermission):
-    pass
-
-
-class BaseCommentView(FormView, DABResponseData):
-    form_class = CommentForm
-
-    def get_context_data(self, **kwargs):
-        context = super().get_context_data(**kwargs)
-        context['comment_form'] = context.pop('form')
-        # context.update(get_comment_context_data(self.request))
-        context.update(DABContext(self.request))
-        return context
-
-    def get_form_kwargs(self):
-        kwargs = super().get_form_kwargs()
-        kwargs['request'] = self.request
-        return kwargs
+    def dispatch(self, request, *args, **kwargs):
+        if not self.has_permission(request):
+            data = {'status': self.status, 'reason': self.reason}
+            return UTF8JsonResponse(status=self.status, data=data)
+        return super().dispatch(request, *args, **kwargs)
 
 
-class CommentCreateMixin(BaseCommentView):
-    email_service = None
+class UserPermission(BasePermission):
+    reason = BlockedUserMSG.NOT_PERMITTED
 
-    def _initialize_email_service(self, comment, request):
-        self.email_service = DABEmailService(comment, request)
+    def has_permission(self, request):
+        return bool(
+            not settings.COMMENT_ALLOW_BLOCKING_USERS or
+            not BlockedUser.objects.is_user_blocked(request.user.id, request.POST.get('email'))
+        )
```

Comment:
i think this `class`is same as the one used in the `permissions` module. https://github.com/Radi85/Comment/pull/201/files#diff-150a36628526769f378f79af3ad910fb6d2e0d6c6ddf28c8cc3a20c8f5d8ba63R27-R35

I think you can probably remove one of these.